### PR TITLE
Daff Callout Component

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/daffio/src/app/content/homepage/component/homepage.component.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    grid-template-columns: 1fr 1fr;
+    padding: 75px 0;
+  }
+
+  &__image {
+    display: block;
+    max-width: 100%;
+  }
+
+  &__list {
```
Comment (damienwebdev) [inline apps/daffio/src/app/content/homepage/component/homepage.component.scss]: Unused.

File: libs/design/src/molecules/callout/callout-body/callout-body.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { DaffCalloutBodyComponent } from './callout-body.component';
+ describe('DaffCalloutBodyComponent', () => {
+  it('should create an instance', () => {
+    const component = new DaffCalloutBodyComponent();
+    expect(component).toBeTruthy();
+  });
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout-body/callout-body.component.spec.ts]: Needs a test for class.

File: libs/design/src/molecules/callout/callout-title/callout-title.component.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+@Component({
+  selector: '[daff-callout-title]',
+  template: '<ng-content></ng-content>',
+  host: {'class': 'daff-callout__title'}
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout-title/callout-title.component.ts]: Needs a test.

File: libs/design/src/molecules/callout/callout-title/callout-title.component.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+})
+export class DaffCalloutTitleComponent {
+   constructor() { }
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout-title/callout-title.component.ts]: Remove empty constructor.

File: libs/design/src/molecules/callout/callout-body/callout-body.component.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+ @Component({
+  selector: '[daff-callout-body]',
+  template: '<ng-content></ng-content>',
+  host: {'class': 'daff-callout__body'}
+})
+export class DaffCalloutBodyComponent {
+   constructor() { }
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout-body/callout-body.component.ts]: Remove empty constructor.

File: libs/design/src/molecules/callout/callout.component.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<div class="daff-callout__container">
+  <ng-content></ng-content>
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout.component.html]: This should probably define some order of title/body.

File: libs/design/src/molecules/callout/callout.component.scss [hunk 7]
```diff
@@ -0,0 +1,152 @@
+@import '../../scss/daff-util';
+
+@mixin callout-headings($titleSize: 2.5rem, $bodySize: 1rem) {
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout.component.scss]: What's the point of this mixin?

File: libs/design/src/molecules/callout/callout.component.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  selector: 'daff-callout',
+  templateUrl: './callout.component.html',
+  styleUrls: ['./callout.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    class: 'daff-callout',
+    '[class.daff-callout--centered]':'layout === "centered"',
+    '[class.daff-callout--grid]':'layout === "grid"',
+    '[class.daff-callout--small]':'size === "small"',
+    '[class.daff-callout--uncontained]':'size === "uncontained"',
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout.component.ts]: I don't think we need a contained variant.

File: libs/design/src/molecules/callout/callout.component.scss [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+.daff-callout {
+  $root: &;
+  @include callout-headings();
+  display: flex;
+  align-items: center;
+  flex-wrap: wrap;
+
+  &__container {
```
Comment (damienwebdev) [inline libs/design/src/molecules/callout/callout.component.scss]: Can we re-use daff--container?

File: libs/design/src/molecules/callout/callout.component.scss (hunk 7 above)
Comment (xelaint) [reply libs/design/src/molecules/callout/callout.component.scss]: Not sure, I replicated what we did on Graycore. I don't remember if you had a reasoning for this.

File: libs/design/src/molecules/callout/callout-body/callout-body.component.ts (hunk 5 above)
Comment (xelaint) [reply libs/design/src/molecules/callout/callout-body/callout-body.component.ts]: Resolved in 463d405

File: libs/design/src/molecules/callout/callout-title/callout-title.component.ts (hunk 4 above)
Comment (xelaint) [reply libs/design/src/molecules/callout/callout-title/callout-title.component.ts]: Resolved in 463d405

File: libs/design/src/molecules/callout/callout-title/callout-title.component.ts (hunk 3 above)
Comment (xelaint) [reply libs/design/src/molecules/callout/callout-title/callout-title.component.ts]: Resolved in 463d405

File: libs/design/src/molecules/callout/callout-body/callout-body.component.spec.ts (hunk 2 above)
Comment (xelaint) [reply libs/design/src/molecules/callout/callout-body/callout-body.component.spec.ts]: Resolved in 463d405

File: libs/design/src/molecules/callout/callout.component.html (hunk 6 above)
Comment (xelaint) [reply libs/design/src/molecules/callout/callout.component.html]: Resolved in e7ff18d

File: libs/design/src/molecules/callout/callout.component.scss (hunk 7 above)
Comment (damienwebdev) [reply libs/design/src/molecules/callout/callout.component.scss]: If it's not used more than once, remove and merge into the class.

File: libs/design/src/molecules/callout/callout.component.scss (hunk 9 above)
Comment (damienwebdev) [reply libs/design/src/molecules/callout/callout.component.scss]: Hrm, then maybe callout__container is a bad name? I think we should use the daff--container as much as possible.

File: libs/design/src/molecules/callout/callout.component.ts (hunk 8 above)
Comment (damienwebdev) [reply libs/design/src/molecules/callout/callout.component.ts]: See my comments on daff-container.

**Pull request merged**